src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointEvidence.java

```diff
@@ -101,71 +96,26 @@ protected BreakpointEvidence( final Kryo kryo, final Input input ) {
      * used by the MapPartitioner to flush pending evidence in the FindBreakEvidenceSpark.WindowSorter.
      */
     public BreakpointEvidence( final int contigIndex ) {
-        this.contigIndex = contigIndex;
+        this.eventLocation = new SVInterval(contigIndex, 0, 0);
         this.templateName = "sentinel";
         this.templateEnd = TemplateEnd.PAIRED_UNKNOWN;
-        this.eventStartPosition = 0;
-        this.eventWidth = 0;
     }
 
     protected void serialize( final Kryo kryo, final Output output ) {
-        output.writeInt(contigIndex);
-        output.writeInt(eventWidth);
-        output.writeInt(eventStartPosition);
+        output.writeInt(eventLocation.getContig());
+        output.writeInt(eventLocation.getStart());
+        output.writeInt(eventLocation.getEnd());
         output.writeString(templateName);
         output.writeByte(templateEnd.ordinal());
     }
 
-    public int getContigIndex() { return contigIndex; }
-    public int getEventStartPosition() { return eventStartPosition; }
-    public int getEventWidth() { return eventWidth; }
-    public int getContigEnd() { return eventStartPosition +eventWidth; }
+    public SVInterval getLocation() { return eventLocation; }
     public String getTemplateName() { return templateName; }
     public TemplateEnd getTemplateEnd() { return templateEnd; }
 
     @Override
-    public int compareTo( final BreakpointEvidence that ) {
-        if ( this == that ) return 0;
-        int result = Integer.compare(this.contigIndex, that.contigIndex);
-        if ( result == 0 ) {
-            result = Integer.compare(this.eventStartPosition, that.eventStartPosition);
-            if ( result == 0 ) {
-                result = Integer.compare(this.eventWidth, that.eventWidth);
-                if ( result == 0 ) {
-                    result = this.templateEnd.compareTo(that.templateEnd);
-                    if ( result == 0 ) {
-                        result = this.templateName.compareTo(that.templateName);
-                        if ( result == 0 ) {
-                            result = this.getClass().getName().compareTo(that.getClass().getName());
-                        }
-                    }
-                }
-            }
-        }
-        return result;
-    }
-
-    @Override
     public String toString() {
-        return contigIndex + "[" + eventStartPosition + ":" + getContigEnd() + "] " + templateName + templateEnd;
-    }
-
-    @Override
-    public boolean equals( final Object obj ) {
```

Comment:
It's very hard to write proper comparison/equality functions for hierarchies in which sub-classes add aspect.  Some argue that it's impossible.  I had been using BreakpointEvidence instances as keys in a sorted tree and needed to hack something that worked, but it wasn't pretty.
I now feel that the whole mess was created by a misanalysis:  By including genomic location fields directly in the class, I was, in effect, asserting that BreakpointEvidence IsA location.  Things have gotten cleaner because now BreakpointEvidence HasA location, and BreakpointEvidence is no longer used as a key.
I eliminated what I felt was an ugly hack.